### PR TITLE
switch to node-rdkafka

### PR DESCRIPTION
**Pull request by jonathan-gramain**
kafka-node has issues dealing with reconnections etc. especially when
    there are concurrent joins of multiple consumers of the same
    group. This often leads to multiple consumers processing the same
    partition, which is not just suboptimal, but breaks lifecycle
    processing.
    
    node-rdkafka is a more maintained and better tested library, relying
    on librdkafka underneath.

### CODE REVIEW

**Comment by jonathan-gramain** [comment]
@ironman-machine 
DEFAULT_BRANCH="rel/7.4"
SCALITY_INTEGRATION_BRANCH="ultron/rel/7.4"
SCALITY_FEDERATION_BRANCH="fwd/7.4/ft/S3C-1154-lifecycle"
try

**Comment by jonathan-gramain** [comment]
@ironman-machine
DEFAULT_BRANCH=rel/7.4
SCALITY_INTEGRATION_BRANCH=ultron/rel/7.4
SCALITY_FEDERATION_BRANCH=fwd/7.4/ft/S3C-1154-lifecycle
try

File: package.json [hunk 1]
```diff
@@ -41,6 +41,7 @@
     "eslint-plugin-react": "^4.2.3",
     "joi": "^10.6",
     "kafka-node": "^2.2.0",
```
Comment (bennettbuchanan) [inline package.json]: Can remove this and update the reference of kafka-node to node-rdkafka in DESIGN.md.

**Comment by jonathan-gramain** [comment]
@ironman-machine
DEFAULT_BRANCH=rel/7.4
SCALITY_INTEGRATION_BRANCH=ultron/rel/7.4
SCALITY_FEDERATION_BRANCH=ft/S3C-1154-lifecycle
try

**Comment by jonathan-gramain** [comment]
@ironman-machine
DEFAULT_BRANCH=rel/7.4
SCALITY_FEDERATION_BRANCH=fwd/7.4/ft/S3C-1154-lifecycle
try

File: extensions/replication/replicationStatusProcessor/ReplicationStatusProcessor.js [hunk 2]
```diff
@@ -92,30 +92,30 @@ class ReplicationStatusProcessor {
...
-            zookeeper: { connectionString: this.zkConfig.connectionString },
+            kafka: { hosts: this.kafkaConfig.hosts },
             topic: this.repConfig.replicationStatusTopic,
             groupId: this.repConfig.replicationStatusProcessor.groupId,
             concurrency:
             this.repConfig.replicationStatusProcessor.concurrency,
             queueProcessor: this.processKafkaEntry.bind(this),
             fetchMaxBytes: CONSUMER_FETCH_MAX_BYTES,
-            createConsumer: (!options || options.createConsumer),
+            bootstrap: true,
```
Comment (philipyoo) [inline extensions/replication/replicationStatusProcessor/ReplicationStatusProcessor.js]: Is this temporary?

File: package.json (hunk 1 above)
Comment (jonathan-gramain) [reply package.json]: thanks, updated

**Comment by jonathan-gramain** [comment]
@ironman-machine try

File: extensions/replication/replicationStatusProcessor/ReplicationStatusProcessor.js (hunk 2 above)
Comment (jonathan-gramain) [reply extensions/replication/replicationStatusProcessor/ReplicationStatusProcessor.js]: No, it's a rework of the way we bootstrap consumers for tests. Now only the bootstrap option is passed, no need to call the `bootstrap()` function later, called internally (and renamed to `_bootstrapConsumer()`).

Comment (jonathan-gramain) [reply extensions/replication/replicationStatusProcessor/ReplicationStatusProcessor.js]: Oh I see what you mean, why is this true in non-test code, I have to check :)

Comment (jonathan-gramain) [reply extensions/replication/replicationStatusProcessor/ReplicationStatusProcessor.js]: Fixed, this was an overlook, thanks for catching it!

Comment (philipyoo) [reply extensions/replication/replicationStatusProcessor/ReplicationStatusProcessor.js]: Ah I see, I wasn't too sure so decided to ask 👍

**Comment by jonathan-gramain** [comment]
@ironman-machine 
DEFAULT_BRANCH=rel/7.4-beta
SCALITY_FEDERATION_BRANCH=tmp/lifecycle-to-7.4-beta
RUN_BACKBEAT_CRR_TESTS=1
try

**Comment by jonathan-gramain** [comment]
@ironman-machine
DEFAULT_BRANCH=rel/7.4-beta
SCALITY_FEDERATION_BRANCH=rel/7.4-beta
RUN_BACKBEAT_CRR_TESTS=1
try

**Comment by jonathan-gramain** [comment]
@ironman-machine
DEFAULT_BRANCH=rel/7.4-beta
SCALITY_INTEGRATION_BRANCH=rel/7.4-beta
RUN_BACKBEAT_CRR_TESTS=1
try

**Comment by jonathan-gramain** [comment]
@ironman-machine
DEFAULT_BRANCH=rel/7.4-beta
SCALITY_INTEGRATION_BRANCH=ultron/rel/7.4-beta
RUN_BACKBEAT_CRR_TESTS=1
try

**Comment by jonathan-gramain** [comment]
@ironman-machine try

**Comment by jonathan-gramain** [comment]
@ironman-machine r+

**Pull request merged**